demux/demux_lavf.c

```diff
@@ -750,14 +757,29 @@ static void handle_new_stream(demuxer_t *demuxer, int i)
         sh->codec->par_w = st->sample_aspect_ratio.num;
         sh->codec->par_h = st->sample_aspect_ratio.den;
 
+#if LIBAVFORMAT_VERSION_INT >= AV_VERSION_INT(60, 15, 100)
+        const AVPacketSideData *sdp = av_packet_side_data_get(st->codecpar->coded_side_data,
+                                                              st->codecpar->nb_coded_side_data,
+                                                              AV_PKT_DATA_DISPLAYMATRIX);
+        uint8_t *sd = sdp ? sdp->data : NULL;
+#else
         uint8_t *sd = av_stream_get_side_data(st, AV_PKT_DATA_DISPLAYMATRIX, NULL);
+#endif
         if (sd) {
-            double r = av_display_rotation_get((uint32_t *)sd);
+            double r = av_display_rotation_get((int32_t *)sd);
             if (!isnan(r))
                 sh->codec->rotate = (((int)(-r) % 360) + 360) % 360;
         }
 
-        if ((sd = av_stream_get_side_data(st, AV_PKT_DATA_DOVI_CONF, NULL))) {
+#if LIBAVFORMAT_VERSION_INT >= AV_VERSION_INT(60, 15, 100)
+        sdp = av_packet_side_data_get(st->codecpar->coded_side_data,
+                                      st->codecpar->nb_coded_side_data,
+                                      AV_PKT_DATA_DOVI_CONF);
+        sd = sdp ? sdp->data : NULL;
+#else
+        sd = av_stream_get_side_data(st, AV_PKT_DATA_DOVI_CONF, NULL);
+#endif
```

Comment:
time for a helper function that wraps this IMO

Reply:
I wasn't sure it is worth for two uses, but I did wrapper for libplacebo, so we can have it here too.